pkg/storage/replica_follower_read.go

```diff
@@ -66,6 +66,12 @@ func (r *Replica) canServeFollowerRead(
 					r.store.cfg.ClosedTimestamp.Storage.(*ctstorage.MultiStorage).StringForNodes(lErr.LeaseHolder.NodeID),
 				)
 			}
+			// We may be able to serve the read based on the lease start time.
+			// It is safe to serve a read for requests which occur before this
+			// Replica's view of the lease start time.
```

Comment:
> This is because anything proposed by a previous leaseholder and yet unobserved won't apply under this new lease, and under this lease nothing can be proposed that would affect timestamps earlier than the start time of the lease (in practice, this is enforced by the timestamp cache on the leaseholder).

The one concern that I have (which is probably not a real one but needs to be sleuthed) is that of lease extensions and how they affect the `Start` timestamp. We basically want `Start` to never move forward during the tenure of a single leaseholder (as measured via the lease sequence number).

It also seems that this works for expiration-based leases as well, so we do seem to have some kind of follower reads for expiration-based ranges, even though it's a pretty shitty one since the Start timestamp won't move forward.